### DISCUSSION TITLE
🔥🔥🔥 v3.2 - Update ... what we are currently working on

### DISCUSSION THREAD

**Discussion by Blackymas** [Blackymas/NSPanel_HA_Blueprint#384] [Announcements]
Hi @ all

as always, a quick update on what we're working on.

The last weeks we have been working hard on some features again.

Unfortunately with the update of HA and Esphome there were a few things we had to change and cost us a lot of time (some small changes are already in beta).

Since this took us a lot of time we need a moment until update v3.2 is ready. 

Unfortunately we had to push the Climate and Mediaplayer changes into the v3.3 release.

&nbsp;

**Nevertheless you can expect many new things in version 3.2**

some of them are:

- multiclimate -> You can now put climates on the buttons. With a "short press" you will get to the Climate page.

- Long Press for HW Buttons. here you can choose if a default action like Lightsettings page should be called or even create your own configuration.

- For all buttons you can now assign your own icons. Also the color of the icons on the button you can customize. So you can customize the button for you, as you need it.

- 7 new chips have been added to the homepage. You can assign the entity and icons freely.

- Hotwater and Humidity fields on the homepage have been deleted and replaced by 3 new free assignable fields. here you can also assign entity and icon freely

- Homepage will be completely customizable in terms of all icons, entities and even the color for icons and labels can be freely assigned.

With the update 3.2 you can build your own homepage as you like it.

I think here again a mega update awaits you ;) 

**As always, here is the hint that you are welcome to support us** 

[![Paypal](https://user-images.githubusercontent.com/41958506/212499642-b2fd097a-0938-4bfc-b37b-74df64592c58.png)](https://www.paypal.com/donate/?hosted_button_id=S974SWQMB8PB2)

![new Home page](https://user-images.githubusercontent.com/41958506/222243452-5fa3b91e-cd0f-4027-aa11-26091e5d07fa.jpg)
![new Buttons](https://user-images.githubusercontent.com/41958506/222243474-201af500-216d-4315-9900-dcb909911ba3.jpg)
![Bildschirmfoto 2023-03-01 um 20 14 53](https://user-images.githubusercontent.com/41958506/222243537-a442a9c1-7219-4116-b984-199fbaad33e7.png)

**Comment by T3mplate-wow** [comment]
Sounds awesome. To be honest I am totally happy with 3.1 and I thought there is no real need for an update. But the ideas/features here are great. 
Just to make sure: changing the background image is not in planning for 3.2. right?

**Reply by Blackymas** [reply]
via Blueprint will unfortunately never work from a technical point of view.

But with 3.2 it will be possible to change the background via HMI because all icons and black bars have been replaced by the font.

You can then insert your image via HMI and create a new TFT.

**Comment by sneekerzzz** [comment]
How can i test this beta ? What are the steps to update etc

**Comment by kroon040** [comment]
Look very nice. Any idea when it will be released?

**Reply by Blackymas** [reply]
it is in beta and is being checked by some to see if everything is ok

**Reply by pwirkus** [reply]
hello , where can i find beta 3.2 ?

**Comment by edwardtfn** [comment]
I'm trying v3.2 with a US panel and couldn't find that top bar (I'm mostly interested in the settings button). 
![image](https://user-images.githubusercontent.com/94725493/225135958-ebbd0426-fe23-413c-aafe-c3a96b7d8886.png)

I can confirm both the TFT, ESPHome and Blueprint are all v3.2.
Am I missing some setting?

**Reply by Blackymas** [reply]
add me in discord I will you add in the us tft group
`Blackymas#0129`

**Comment by cclarke99** [comment]
Outstanding stuff coming! Are there any plans to allow region customizations? Particularly interested in the ability to have 12 or 24 hour clock and have the date strings formatted for the US.

**Comment by robgough1970** [comment]
is there an ETA for when 3.2 will be released .. had lots of trouble getting my panels to work for various reasons. I am now up and running but not configured pages etc yet was waitimg for 3.2 incase i had to start over again.

**Reply by sneekerzzz** [reply]
I started using the beta and for my case it's just works fine. So maybe try

The only thing I want to know is where to find color codes

**Reply by robgough1970** [reply]
thanks... ive been looking for a link for 3.2 but couldn't find it ... care to share where it is

**Reply by edwardtfn** [reply]
v3.2 is in the `beta` branch in the repo.

**Reply by sneekerzzz** [reply]
Switch branches to Beta, download the zip and edit/swap all the necessary files

**Reply by robgough1970** [reply]
cheers ... will put it to the test

**Reply by edwardtfn** [reply]
> The only thing I want to know is where to find color codes

I agree the color code will be challenging.
I believe it is just the decimal value of the hex RRGGBB that is much more familiar to most people.

@Blackymas, do you want me to work on supporting both the colors as it is or hexadecimal (RRGGBB) colors? We probably can release 3.2 as it is (decimal only) and work in the support for both in a near future version.

**Reply by edwardtfn** [reply]
> I believe it is just the decimal value of the hex RRGGBB that is much more familiar to most people.

This is not correct. The colors are using rgb565.
You will find converters online.
I will try to look in a way to support the standard 24-bit rgb hex and then convert int he blueprint before sending to the panel.
